cmd/main_test.go

```diff
@@ -301,3 +330,96 @@ var _ = DescribeTable("TestRunAccessCheck", func(user string, namespace string,
 		"patch",
 		false),
 )
+
+var _ = DescribeTable("TestGetNamespacesWithAccess", func(allNamespaces []k8sapi.Namespace,
+	expectedNs []k8sapi.Namespace, actualNs []k8sapi.Namespace, err error) {
+	e := echo.New()
+	cfg, _ := config.GetConfig()
+	clientset, _ := kubernetes.NewForConfig(cfg)
+	req := httptest.NewRequest(http.MethodGet, "/", nil)
+	rec := httptest.NewRecorder()
+	c := e.NewContext(req, rec)
+	c.Request().Header.Set("X-Email", "user1@konflux.dev")
+	authCl := clientset.AuthorizationV1()
+	
+	JustBeforeEach(func() {
+		actualNs, err = getNamespacesWithAccess(e, c, authCl, allNamespaces)
+	})
+
+	Context("When all the namespaces have all the necessary permissions like create, list, watch and delete", func() {
+		namespaceNames := []string{"ns-test-tenant-1", "ns-test-tenant-2"}
+		roleNames := []string{"ns-namespace-access-1", "ns-namespace-access-2"}
+		roleBindingNames := []string{"ns-namespace-access-user-binding-1", "ns-namespace-access-user-binding-2"}
+		BeforeEach(func() {
+			for i, name := range namespaceNames {
+				ns, err := createNamespace(k8sClient, name)
+				Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace %s: %v", name, err))
+				allNamespaces = append(allNamespaces, ns)
+				createRole(k8sClient, name, roleNames[i], []string{"create", "list", "watch", "delete"})
+				createRoleBinding(k8sClient, roleBindingNames[i], name, "user1@konflux.dev", roleNames[i])
+			}			
+			expectedNs = allNamespaces
+		})
+		It("returns all namespaces in the list", func() {
+			Expect(actualNs).To(Equal(expectedNs))
+			Expect(err).NotTo(HaveOccurred(), "Unexpected error testing GetNamespacesWithAccess")
+		})
+		AfterEach(func() {
+			for i, name := range namespaceNames {
+				deleteRoleBinding(k8sClient, name, roleBindingNames[i])
+				deleteRole(k8sClient, name, roleNames[i])
```

Comment:
It works, but I wonder if we can put it in a mapping or a struct to make sure each namespace, role, and rolebinding are indeed related to each other, rather than count on index numbers.

Another approach is creating those  resources the same way we delete them (a loop), as we already have a loop for creating the namespaces. therefore, declaring the alignment between them and the namespaces beforehand.

Reply:
Updated for this test case with struct mapping but for the other test case which was not in the loop I didn't update createRole and createRoleBinding.